packages/core/src/combinator/withArrayValues.js

```diff
@@ -35,11 +36,6 @@ class ZipArrayValuesSink extends Pipe {
 
   event (t, b) {
     const f = this.f
-    this.sink.event(t, f(this.values[this.index], b))
-
-    this.index += 1
-    if (this.index >= this.values.length) {
-      this.sink.end(t)
-    }
+    this.sink.event(t, f(this.values[this.index++], b))
```

Comment:
While we all enjoy one-liners, incrementing `this.index` in the array makes it less clear and is easily overlooked, why I advise having the incrementor on a separate line. Also, there could arise issues of the increment operators position—the off-by-one error—i.e., is the index incremented before or after retrieving the value? (Don't worry, I know).

Consider these:
```
x = a+++b -> x = (a++)+b -> x = a + b; a++ // The tokenizer is greedy.
```

The most important rational, however, is that you are actually causing a side effect besides returning a value, and that is what makes the code harder to reason about.

Reply:
Agreed.  I vote for `+= 1` on the next line.